core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java

```diff
@@ -470,6 +480,76 @@ public void testTasksWaitForAllTask() throws Exception {
         assertThat(response.getTasks().size(), greaterThanOrEqualTo(1));
     }
 
+    public void testTaskResultPersistence() throws Exception {
+        // First make sure that template was correctly created
+        GetIndexTemplatesResponse response =
+            client().admin().indices().prepareGetTemplates(TaskResultsService.TASK_RESULT_INDEX_TEMPLATE).get();
+        assertEquals(1, response.getIndexTemplates().size());
+
+        registerTaskManageListeners(TestTaskPlugin.TestTaskAction.NAME);  // we need this to get task id of the process
+
+        // Start non-blocking test task
+        TestTaskPlugin.TestTaskAction.INSTANCE.newRequestBuilder(client()).setShouldPersistResult(true).setShouldBlock(false).get();
```

Comment:
I made that comment when I was thinking of the feature in a broader, maybe misguided sense. It isn't valid given the conversation we had about this only being a thing with `wait_for_completion=false`.